File: sommelier-analytics-mvp/backend/app/models/restaurant.py
"""
Restaurant model
"""
from sqlalchemy import Column, String, DateTime, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from datetime import datetime
import uuid
from app.core.database import Base


class Restaurant(Base):
    """Restaurant entity"""
    __tablename__ = "restaurants"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String(255), nullable=False)
    email = Column(String(255), unique=True, nullable=False)
    phone = Column(String(20), nullable=True)
    address = Column(String(500), nullable=True)
    city = Column(String(100), nullable=True)
    state = Column(String(50), nullable=True)
    zip_code = Column(String(10), nullable=True)
    
    # Subscription status
    is_active = Column(Boolean, default=True)
    subscription_tier = Column(String(50), default="trial")  # trial, basic, pro, enterprise
    
    # Timestamps
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    
    # Relationships
    wines = relationship("Wine", back_populates="restaurant", cascade="all, delete-orphan")
    sales = relationship("Sale", back_populates="restaurant", cascade="all, delete-orphan")
    dishes = relationship("Dish", back_populates="restaurant", cascade="all, delete-orphan")
    
    def __repr__(self):
        return f"<Restaurant {self.name}>"

File: sommelier-analytics-mvp/frontend/src/types/restaurant.ts
import { z } from 'zod';

export const restaurantSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  email: z.string().email(),
  phone: z.string().nullable(),
  address: z.string().nullable(),
  city: z.string().nullable(),
  state: z.string().nullable(),
  zip_code: z.string().nullable(),
  is_active: z.boolean(),
  subscription_tier: z.string(),
});

export const restaurantFormSchema = z.object({
  name: z.string().min(1, 'Restaurant name is required'),
  email: z.string().email('Valid email is required'),
  phone: z.string().optional(),
  address: z.string().optional(),
  city: z.string().optional(),
  state: z.string().optional(),
  zip_code: z.string().optional(),
});

export type Restaurant = z.infer<typeof restaurantSchema>;
export type RestaurantFormData = z.infer<typeof restaurantFormSchema>;
